tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/BuilderUtil.scala

```diff
@@ -25,7 +25,7 @@ object BuilderUtil {
 
   /** Removes elem from the scope, as the scope only contains types of free variables */
   def markAsBound(elem: TlaEx): TBuilderInternalState[Unit] = State[TBuilderContext, Unit] { mi: TBuilderContext =>
-    require(elem.isInstanceOf[NameEx])
+    require(elem.isInstanceOf[NameEx], s"elem = $elem must be a variable name.")
```

Comment:
So say that `elem` were incorrectly an integer expression, am I correct in understanding that this error would read `"elem = TlaInt(1) must be a variable name"`? I don't think I'd now how to parse that out of context. I think i'd be looking for a bad assignment or something.

Would it be clearer if the message were just something like:

```suggestion
    require(elem.isInstanceOf[NameEx], s"$elem must be a variable name.")
```